### DISCUSSION TITLE
building-and-asp-net-core-mvc-app-with-npm-and-webpack-asp-net-core-2-0-edition/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#54] [Blog comments]
# building-and-asp-net-core-mvc-app-with-npm-and-webpack-asp-net-core-2-0-edition/

https://blogs.taiga.nl/martijn/posts/building-and-asp-net-core-mvc-app-with-npm-and-webpack-asp-net-core-2-0-edition/

**Comment by disqus-comments-migrator** [comment]
<em>[James 'Steve' Blandford](https://disqus.com/by/jamessteveblandford/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at december 10 2017, 07:59</em>

---

<p>thanks for the quick setup for ASP.NET Core 2.0!! I have started learning C# (coming from a web developer background) and this really helps me link my existing knowledge of NPM with .NET Core</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at december 10 2017, 09:35, in reply to [James 'Steve' Blandford](https://disqus.com/by/jamessteveblandford/)'s [comment](https://github.com/martijnboland/blog/discussions/54#discussioncomment-8953000)</em>

---

<p>Thanks, glad I could help</p>

**Comment by disqus-comments-migrator** [comment]
<em>[bestardnyc](https://disqus.com/by/bestardnyc/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 22 2018, 04:28</em>

---

<p>Great article, I have a question about "Step 4: Bundle with webpack", in the past I add cache busting in webpack by including  [chunkhash] in the name. From your example, it's using "asp-append-version=true", are they the same concept? will it generate the same or different Hash IF the file content has NOT changed?  Thanks for sharing.</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 22 2018, 04:37, in reply to [bestardnyc](https://disqus.com/by/bestardnyc/)'s [comment](https://github.com/martijnboland/blog/discussions/54#discussioncomment-8953003)</em>

---

<p>Based on the info in <a href="https://stackoverflow.com/questions/33342643/how-does-javascript-version-asp-append-version-work-in-asp-net-core-mvc" rel="nofollow noopener" title="https://stackoverflow.com/questions/33342643/how-does-javascript-version-asp-append-version-work-in-asp-net-core-mvc">https://stackoverflow.com/questions/33342643/how-does-javascript-version-asp-append-version-work-in-asp-net-core-mvc</a>, I think it will calculate the same hash when the content has not changed.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Jopo](https://disqus.com/by/disqus_KXT3fdP2Gn/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 19 2018, 06:01</em>

---

<p>One question Martijn, why should one separate vendor and application scripts, instead bundling all into one bundle?</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 20 2018, 03:53, in reply to [Jopo](https://disqus.com/by/disqus_KXT3fdP2Gn/)'s [comment](https://github.com/martijnboland/blog/discussions/54#discussioncomment-8953005)</em>

---

<p>Two reasons: a vendor bundle probably changes less often than the app code, so you can benefit from browser caching and webpack rebuilds are faster when only the app bundle needs a rebuild (at least, this used to be, don't know if that's still the case).</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Jopo](https://disqus.com/by/disqus_KXT3fdP2Gn/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 20 2018, 04:46, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/54#discussioncomment-8953007)</em>

---

<p>Yes, that's what I thought of too, thanks for your confirmation ;-) Keep on the great work!</p>

**Comment by disqus-comments-migrator** [comment]
<em>[teoman shipahi](https://disqus.com/by/aytekinozdemir/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at april 21 2018, 06:51</em>

---

<p>Amazing article! Just one thing, I believe target property should be "node" for webpack config. Because it defaults to web, and creates errors while trying to resolve modules references fs etc.<br><a href="https://webpack.js.org/concepts/targets/" rel="nofollow noopener" title="https://webpack.js.org/concepts/targets/">https://webpack.js.org/concepts/targets/</a></p><p>Also people who are using Bootstrap 4 should consider install and import "popper.js" in main.js.<br>Rest seems to be working fine even with webpack 4 with some minor warnings.</p><p>Thanks!</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at april 22 2018, 09:26, in reply to [teoman shipahi](https://disqus.com/by/aytekinozdemir/)'s [comment](https://github.com/martijnboland/blog/discussions/54#discussioncomment-8953010)</em>

---

<p>Thanks for your comment. I don't however think the target property should be 'node'. And yes, that should create errors when using fs etc. in the browser because those are not available anyway in the browser.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Nick Demis](https://disqus.com/by/nickdemis/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at april 26 2018, 10:26</em>

---

<p>This does not work with bootstrap 4.1 for me. Any suggestions?</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Nick Demis](https://disqus.com/by/nickdemis/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at april 27 2018, 06:48, in reply to [Nick Demis](https://disqus.com/by/nickdemis/)'s [comment](https://github.com/martijnboland/blog/discussions/54#discussioncomment-8953013)</em>

---

<p>ok i had wrog popper.js version i tried popper.js 1.14.3  and work fine!</p>

**Comment by disqus-comments-migrator** [comment]
<em>[dejanmauer](https://disqus.com/by/dejanmauer/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 17 2018, 03:47</em>

---

<p>Great article! However when I try to use jQuery in a view, I always get error "Uncaught ReferenceError: $ is not defined". What am I missing?</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 17 2018, 07:37, in reply to [dejanmauer](https://disqus.com/by/dejanmauer/)'s [comment](https://github.com/martijnboland/blog/discussions/54#discussioncomment-8953017)</em>

---

<p>When you want to use the global $ variabele, you have to expose it with the expose-loader (<a href="https://github.com/webpack-contrib/expose-loader)" rel="nofollow noopener" title="https://github.com/webpack-contrib/expose-loader)">https://github.com/webpack-contrib/expose-loader)</a>. Run `npm install --save-dev expose-loader` and add the following rule to webpack.config.js in the module.rules array:</p><p>{<br>  test: require.resolve('jquery'),<br>  use: [{<br>    loader: 'expose-loader',<br>    options: '$'<br>  }]<br>}</p><p>Also, make sure that any reference to $ happens after your main.js has been loaded.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[dejanmauer](https://disqus.com/by/dejanmauer/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 18 2018, 07:46, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/54#discussioncomment-8953019)</em>

---

<p>This is it! Thank you!</p>
